### ISSUE TITLE
Query incorrectly returns an empty data set during certain times of the day

### ISSUE THREAD

**Issue by bartek-hitachi** [timescale/timescaledb#3428]
**Relevant system information:**
 - OS: Ubuntu 18.04
 - PostgreSQL version: 11.8
 - TimescaleDB version: 1.6.0
 - Installation method: `apt install`

**Describe the bug**
I have a database table called "telemetry". Here are the SQL statements used to create the table:
```
CREATE TABLE IF NOT EXISTS telemetry (
  ts                    TIMESTAMPTZ(3)    NOT NULL,
  station_serialnumber  INTEGER           NOT NULL,
  component_name        TEXT              NOT NULL,
  value                 DOUBLE PRECISION  NULL
);
SELECT create_hypertable('telemetry', 'ts', chunk_time_interval => INTERVAL '3 days');

ALTER TABLE telemetry SET (
  timescaledb.compress,
  timescaledb.compress_segmentby = 'station_serialnumber, component_name'
);
SELECT add_compress_chunks_policy('telemetry', INTERVAL '3 days');
```

The table contains second-by-second data that is being inserted into it almost real-time. And so, the following query normally returns two rows, which is the expected behavior:
```
SELECT
    station_serialnumber AS station,
    component_name AS metric,
    LAST(ts, ts) AS time,
    LAST(value, ts)::integer AS value
FROM telemetry AS T
WHERE
    component_name IN ('PT_SupplyBank1', 'PT_SupplyBank2')
    AND station_serialnumber = 72 AND ts > CURRENT_TIMESTAMP - INTERVAL '1 day'
GROUP BY 1,2
ORDER BY 1,2
```

We run this query on a schedule as part of a reporting module. The issue is that when the query is executed at the start of an hour (e.g. 8:00 AM UTC, 9:00 AM UTC, etc.) it almost always returns empty. That is in spite of the fact that there are plent of records in the telemetry table that meet the criteria in the WHERE clause. When the query is executed again just a minute or two later (or any other time during the hour) it returns two rows as expected.

I configured our reporting module to log the result of an `EXPLAIN ANALYZE` query. Whenever the query returns empty, here is what the `EXPLAIN ANALYZE` looks like:
```
Finalize GroupAggregate  (cost=10893732.72..10893791.13 rows=396 width=31) (actual time=9.911..9.911 rows=0 loops=1)
  Group Key: t.station_serialnumber, t.component_name
  ->  Gather Merge  (cost=10893732.72..10893778.26 rows=396 width=83) (actual time=9.910..10.014 rows=0 loops=1)
        Workers Planned: 1
        Workers Launched: 0
        ->  Sort  (cost=10892732.71..10892733.70 rows=396 width=83) (actual time=0.022..0.022 rows=0 loops=1)
              Sort Key: t.station_serialnumber, t.component_name
              Sort Method: quicksort  Memory: 25kB
              ->  Partial HashAggregate  (cost=10892709.69..10892715.63 rows=396 width=83) (actual time=0.006..0.006 rows=0 loops=1)
                    Group Key: t.station_serialnumber, t.component_name
                    ->  Parallel Custom Scan (ChunkAppend) on telemetry t  (cost=0.57..399071.79 rows=1049363790 width=35) (actual time=0.000..0.001 rows=0 loops=1)
                          Chunks excluded during startup: 43
                          ->  Parallel Index Scan using _hyper_20_560_chunk_telemetry_ts_idx on _hyper_20_560_chunk t_1  (cost=0.57..358248.97 rows=50789 width=35) (never executed)
                                Index Cond: (ts > (CURRENT_TIMESTAMP - '1 day'::interval))
                                Filter: ((component_name = ANY ('{PT_SupplyBank1,PT_SupplyBank2}'::text[])) AND (station_serialnumber = 72))
Planning Time: 140.858 ms
Execution Time: 21.692 ms
```

And when the query correctly returns two rows just a minute later, here is what the `EXPLAIN ANALYZE` looks like:
```
Finalize GroupAggregate  (cost=10893482.78..10893541.19 rows=396 width=31) (actual time=113229.328..113229.335 rows=2 loops=1)
  Group Key: t.station_serialnumber, t.component_name
  ->  Gather Merge  (cost=10893482.78..10893528.32 rows=396 width=83) (actual time=113228.670..113297.115 rows=2 loops=1)
        Workers Planned: 1
        Workers Launched: 1
        ->  Sort  (cost=10892482.77..10892483.76 rows=396 width=83) (actual time=56600.468..56600.468 rows=1 loops=2)
              Sort Key: t.station_serialnumber, t.component_name
              Sort Method: quicksort  Memory: 25kB
              Worker 0:  Sort Method: quicksort  Memory: 25kB
              ->  Partial HashAggregate  (cost=10892459.75..10892465.69 rows=396 width=83) (actual time=56600.424..56600.425 rows=1 loops=2)
                    Group Key: t.station_serialnumber, t.component_name
                    ->  Parallel Custom Scan (ChunkAppend) on telemetry t  (cost=0.57..398822.20 rows=1049363755 width=35) (actual time=0.049..56536.995 rows=28597 loops=2)
                          Chunks excluded during startup: 43
                          ->  Parallel Index Scan using _hyper_20_560_chunk_telemetry_ts_idx on _hyper_20_560_chunk t_1  (cost=0.57..357999.38 rows=50754 width=35) (actual time=0.096..113061.337 rows=57194 loops=1)
                                Index Cond: (ts > (CURRENT_TIMESTAMP - '1 day'::interval))
                                Filter: ((component_name = ANY ('{PT_SupplyBank1,PT_SupplyBank2}'::text[])) AND (station_serialnumber = 72))
                                Rows Removed by Filter: 11934808
Planning Time: 95.594 ms
Execution Time: 113298.975 ms
```

What is surprising to me is that in the former case, the Parallel Index Scan is never executed. Why would that be?

**Expected behavior**
Given that there are always multiple records in the telemetry table that satisfy the WHERE clause, the query should always return some data or at least throw an error if there is an issue.

**Actual behavior**
The query behaves inconsistently. Most of the time it returns two rows as expected, but when executed at the beginning of an hour, it usually returns empty and almost immediately, as if there was something interfering.

**Comment by pmwkaa** [comment]
@bartek-hitachi Thanks for submitting the issue. Do you think a have a chance to test it with the latest TimescaleDB version?

**Comment by bartek-hitachi** [comment]
@pmwkaa we were able to verify that upgrading to TimescaleDB v1.7.5 fixes this issue. We suspect that the issue we were having was a result of this known bug which was fixed in v1.7: https://github.com/timescale/timescaledb/issues/1793

**Comment by pmwkaa** [comment]
@bartek-hitachi Thats good news! I'll close the issue for now, if it pops up let us know
